app/controllers/admin/api/cms/templates_controller.rb

```diff
@@ -132,11 +139,21 @@ def find_template
   end
 
   def cms_template_params
-    params.require(:template).permit(*ALLOWED_PARAMS)
+    params.require(:template).permit(*allowed_type_params)
   end
 
   private
 
+  def allowed_type_params
+    ALLOWED_PARAMS[template_type]
+  end
+
+  def template_type
+    return params[:type].parameterize.to_sym if params[:type].present?
```

Comment:
Just to double check: If an unknown type is provided, `Admin::Api::CMS::TemplateService` will be responsible for raising the error, right? Also, is the error displayed in the API response? I mean, do we have some some rescue to return the error and a 4xx status?

Reply:
yes, an error will be raised from `TemplateService` and captured at [app/controllers/admin/api/cms/templates_controller.rb#L61](https://github.com/3scale/porta/blob/9f628f67d67cbad5b364d4213e44cb1c582076ee/app/controllers/admin/api/cms/templates_controller.rb#L61)